minimesos/src/main/java/com/containersol/minimesos/main/Main.java

```diff
@@ -60,28 +72,30 @@ public void run(String[] args) {
         } catch (Exception e) {
             LOGGER.error("Failed to parse parameters. " + e.getMessage() + "\n");
             printUsage(null);
-            return;
+            return RC_ERR;
         }
 
         if (jc.getParameters().get(0).isAssigned()) {
             printUsage(null);
-            return;
+            return RC_OK;
         }
 
         if (jc.getParsedCommand() == null) {
             MesosCluster cluster = ClusterRepository.loadCluster();
             if (cluster != null) {
                 cluster.printServiceUrls(output);
+                return RC_OK;
             } else {
                 printUsage(null);
+                return RC_ERR;
             }
-            return;
         }
 
         Command parsedCommand = commands.get(jc.getParsedCommand());
 
         if (parsedCommand == null) {
             LOGGER.error("No such command: " + jc.getParsedCommand());
```

Comment:
No, first check if JCommander managed to read the command name. The second check if we have command Object matching the name